src/sagemaker/pipeline.py

```diff
@@ -345,7 +346,7 @@ def register(
                 framework_version=framework_version,
                 nearest_model_name=nearest_model_name,
                 data_input_configuration=data_input_configuration,
-                container_def=container_def,
+                container_list=container_def,
```

Comment:
is it supposed to be a container_list. you are using container_def below

Reply:
Yes, this should be a `container_list`. I used the Pipeline variable test mechanism (pending internal team review in https://github.com/aws/sagemaker-python-sdk/pull/3163) to scan through, and found out that the `container_def=container_def,` would cause a bug.

The reason is PipelineModel here is actually a list of models and `container_def` generated by below is actually a list of container definitions
```
container_def = self.pipeline_container_def(
                inference_instances[0] if inference_instances else None
            )
```

Therefore, if we assign the list of container_def to container_def (like previously), it would break [here](https://github.com/qidewenwhen/sagemaker-python-sdk/blob/345033bb9ab393570856ed48dba937d04cf69f78/src/sagemaker/utils.py#L800) with error `AttributeError: 'list' object has no attribute 'update'`